framework/include/utils/InputParameters.h

```diff
@@ -600,24 +600,37 @@ class InputParameters : public Parameters
    * Get the default value for a postprocessor added with addPostprocessor
    * @param name The name of the postprocessor
    * @param suppress_error If true, the error check is suppressed
+   * @param index The index in the default postprocessor vector
    * @return The default value for the postprocessor
    */
   const PostprocessorValue & getDefaultPostprocessorValue(const std::string & name,
-                                                          bool suppress_error = false) const;
+                                                          bool suppress_error = false,
+                                                          unsigned int index = 0) const;
+
+  /**
+   * Reserve space for default postprocessor values
+   * @param name The name of the postprocessor
+   * @param size Number of entries required in default p
+   */
+  void reserveDefaultPostprocessorValueStorage(const std::string & name, unsigned int size);
```

Comment:
This particular method should probably be private. I don't see a reason to publicly expose it.

Reply:
Same here: How would the parser use it then?

Reply:
Ah, so you do call this externally. I missed that. It still arguably is not a great public method but I can live with it. An alternative is to go declare just this one (or two) method(s) as friends in the one place you need to use them externally. It helps cut down on our public API, which we need to be watching as our project grows.